src/createEventEmitterProxy.test.js

```diff
@@ -0,0 +1,159 @@
+const EventEmitter = require('events');
+
+const { createEventEmitterProxy } = require('.');
+
+describe('createEventEmitterProxy', () => {
+  it('migrates event listeners when switching the target', () => {
```

Comment:
The test names weren't as descriptive as they could be, so I did my best to guess the intent of the test. We should probably rewrite these at some point to be a little more focused, to use `jest.fn()` and mock function assertions instead of manually incrementing a counter when a function is called, etc.